src/parser.cpp

```diff
@@ -364,7 +364,7 @@ namespace Sass {
   Definition_Obj Parser::parse_definition(Definition::Type which_type)
   {
     Scope parent = stack.empty() ? Scope::Rules : stack.back();
-    if (parent != Scope::Root && parent != Scope::Rules && parent != Scope::Function) {
+    if (parent != Scope::Root && parent != Scope::Rules && parent != Scope::Function && parent != Scope::Media) {
```

Comment:
I'm looking into why the check_nesting visitor is bailing out early here. It's a direct port from Ruby Sass so tracking down the issue should be fast.